core/build_playerglobal/src/lib.rs

```diff
@@ -20,6 +20,9 @@ const RUFFLE_METADATA_NAME: &str = "Ruffle";
 // Indicates that we should generate a reference to an instance allocator
 // method (used as a metadata key with `Ruffle` metadata)
 const METADATA_INSTANCE_ALLOCATOR: &str = "InstanceAllocator";
+// Indicates that we should generate a reference to a native initializer
+// method (used as a metadata key with `Ruffle` metadata)
+const METADATA_NATIVE_INSTANCE_INIT: &str = "NativeInstanceInit";
```

Comment:
alternatively, we could set some flag on the class, to mark it non-user-instantiatable.
(IIRC this would also apply to a bunch of other classes, like System/Keyboard (which don't need to be instantiated at all, even natively) or MorphShape)

Reply:
This also feels better in the sense that we might want to construct an object from native code and have it run a normal AS constructor, while disallowing constructing it by user code.

Reply:
> This also feels better in the sense that we might want to construct an object from native code and have it run a normal AS constructor, while disallowing constructing it by user code.

The only reason I could see for doing that would be if we wanted to invoke ActionScript bytecode. This could be accomplished by defining something like `private function init() { ... }` in the ActionScript file, and then delegating to that from the native constructor.

Having playerglobal support for native instance initializers will also be useful for porting existing classes to ActionScript, so I don't think we need an additional flag mechanism.

Reply:
Related note: our notion of native and regular initializers is a little misnamed. Native initializers are actually callable by AS3: specifically, they get called by `constructsuper` and friends. It's how we prevent `DisplayObject` from being constructed but still allow it to be subclassed. If you want to forbid subclassing you'd also have to check the type of `this`.

The most robust approach would be to have `#[Ruffle(NativeInstanceInit)]` attach to static methods to nominate them as the native initializer for this type, that would let us have AS3 impls of native inits. But the current approach of putting it on the class to ask for a Rust impl is good enough that I'm willing to merge with what we have.